uaclient/jobs/gcp_auto_attach.py

```diff
@@ -0,0 +1,36 @@
+"""
+Try to auto-attach in a GCP instance. This should only work
+if the instance has a new UA license attached to it
+"""
+
+from uaclient import config
+from uaclient import exceptions
+
+from uaclient.clouds.identity import get_cloud_type
+from uaclient.cli import action_auto_attach
+
+
+def gcp_auto_attach(cfg: config.UAConfig) -> None:
+    # If the instance is already attached we will not do anything.
+    # This implies that the user may have a new license attached to the
+    # instance, but we will not perfom the change through this job.
+    if cfg.is_attached:
+        return
+
+    # We will not do anything in a non-GCP cloud
+    cloud_id, _ = get_cloud_type()
+    if not cloud_id or cloud_id != "gce":
+        return
+
+    try:
+        # This function already uses the assert lock decorator,
+        # which means that we don't need to make create another
+        # lock only for the job
+        action_auto_attach(args=None, cfg=cfg)
```

Comment:
I don't know if we should reuse here the direct cli command for auto-attach, but I think this is a more straightforward solution

Reply:
I see no problem in that